File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  let indices = [];
  for (let i = 0; i < nums.length; i++) {
    let newNums = nums.toSpliced(i, 1);
    newNums.forEach(n => {
      if (n + nums[i] === target) {
        !indices.includes(i) && indices.push(i);
        !indices.includes(nums.indexOf(n)) && indices.push(nums.indexOf(n));
      }
    });
  }
  return indices;
};

console.log(twoSum([3, 2, 4], 6));

File: 1-two-sum/two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dict  = new Dictionary<int, int>();
        for (int i = 0; i < nums.Length; i++) {
            dict[nums[i]] = i;
        }

        for (int i = 0; i < nums.Length; i++) {
            int part = target - nums[i];
            if (dict.ContainsKey(part) && dict[part] != i) {
                return new int[]{i, dict[part]};
            }
        }
        
        return new int[]{};
    }
}